src/QsCompiler/Transformations/MonomorphizationValidation.cs

```diff
@@ -94,10 +87,10 @@ private class TypeTransformation : TypeTransformation<TransformationState>
         {
             public TypeTransformation(SyntaxTreeTransformation<TransformationState> parent) : base(parent, TransformationOptions.NoRebuild) { }
 
-            public override QsTypeKind<ResolvedType, UserDefinedType, QsTypeParameter, CallableInformation> OnTypeParameter(QsTypeParameter tp)
-            {
-                throw new Exception("Type Parameter types must be resolved");
-            }
+            public override QsTypeKind<ResolvedType, UserDefinedType, QsTypeParameter, CallableInformation> OnTypeParameter(QsTypeParameter tp) =>
+                SharedState.IntrinsicCallableSet.Contains(tp.Origin) 
```

Comment:
I am not sure why this would be needed. The only time type parameters would come up would be in the implementations of exempt callables, and those are either intrinsic, or not type-parameterized. Could you point out when this logic would be needed?

Reply:
If an intrinsic callable is generic, then its type parameters can occur within expressions.

Reply:
We had discussed this offline, and you had given the explanation that when generic intrinsics are called in the code, the type of that call contains type parameter types. Please add a comment to the code here with that expatiation to help make this clear for future readings of this.

Reply:
done